File: src/dataset/go/rna_transcription_Exercism_go_rootulp/openAI/ChatGPT4_rna_transcription_v4.go
package strand

func ToRNA(dna string) string {
	rna := make([]byte, len(dna))
	for i, r := range dna {
		switch r {
		case 'C':
			rna[i] = 'G'
		case 'G':
			rna[i] = 'C'
		case 'T':
			rna[i] = 'A'
		case 'A':
			rna[i] = 'U'
		}
	}
	return string(rna)
}

File: src/dataset/c/rna-transcription_exercism-leleah/openAI/ChatGPT4_rna_transcription_v4.c
#include "rna_transcription.h"

char *to_rna(const char *dna)
{
    if (!dna) return NULL;

    size_t len = strlen(dna);
    char *rna = malloc(len + 1);
    if (!rna) return NULL;

    for (size_t i = 0; i < len; i++) {
        switch (dna[i]) {
            case 'G': rna[i] = 'C'; break;
            case 'C': rna[i] = 'G'; break;
            case 'T': rna[i] = 'A'; break;
            case 'A': rna[i] = 'U'; break;
            default: free(rna); return NULL; // Handle invalid input
        }
    }
    rna[len] = '\0';
    return rna;
}